### PR TITLE
Introduce Larastan

### PR DESCRIPTION
**Pull request by szepeviktor**
Hello @DarkGhostHunter!

This PR introduces static analysis.
If you are interested in its benefits I'm open to talk about these issues.

### CODE REVIEW

**Comment by szepeviktor** [comment]
<details><summary>Current issues</summary>

```
 ------ -------------------------------------------------------------------------------------------------------------
  Line   Assertion/Creator/Pipes/MayRetrieveCredentialsIdForUser.php
 ------ -------------------------------------------------------------------------------------------------------------
  43     Method Laragear\WebAuthn\Assertion\Creator\Pipes\MayRetrieveCredentialsIdForUser::parseCredentials() should
         return Illuminate\Support\Collection<int, array> but returns Illuminate\Support\Collection<int, array{id?:
         mixed, type: string, transports?: non-empty-array<int, string>}>.
 ------ -------------------------------------------------------------------------------------------------------------

 ------ ---------------------------------------------------------------------------------------------------------------
  Line   Assertion/Validator/Pipes/CheckCredentialIsForUser.php
 ------ ---------------------------------------------------------------------------------------------------------------
  59     Parameter #1 $model of method
         Illuminate\Database\Eloquent\Relations\BelongsTo<Laragear\WebAuthn\Contracts\WebAuthnAuthenticatable,Laragear
         \WebAuthn\Models\WebAuthnCredential>::isNot() expects Illuminate\Database\Eloquent\Model|null,
         Laragear\WebAuthn\Contracts\WebAuthnAuthenticatable|null given.
 ------ ---------------------------------------------------------------------------------------------------------------

 ------ ----------------------------------------------------------------------------------------------------------
  Line   Attestation/AuthenticatorData.php
 ------ ----------------------------------------------------------------------------------------------------------
  82     PHPDoc tag @param has invalid value (object{aaguid: int|bool, credentialId: string, credentialPublicKey:
         string}  $attestedCredentialData): Unexpected token "{", expected variable at offset 205
 ------ ----------------------------------------------------------------------------------------------------------

 ------ -----------------------------------------------------------------------------------------------------
  Line   Auth/WebAuthnUserProvider.php
 ------ -----------------------------------------------------------------------------------------------------
  49     Parameter #2 $autoload of function class_implements expects bool, string given.
  51     Method Laragear\WebAuthn\Auth\WebAuthnUserProvider::retrieveByCredentials() should return
         Illuminate\Contracts\Auth\Authenticatable|null but returns Illuminate\Database\Eloquent\Model|null.
 ------ -----------------------------------------------------------------------------------------------------

 ------ -------------------------------------------------------------------------------------------------------------
  Line   ByteBuffer.php
 ------ -------------------------------------------------------------------------------------------------------------
  407    Method Laragear\WebAuthn\ByteBuffer::fromBase64Url() should return static(Laragear\WebAuthn\ByteBuffer) but
         returns Laragear\WebAuthn\ByteBuffer.
  418    Strict comparison using === between false and string will always evaluate to false.
  422    Method Laragear\WebAuthn\ByteBuffer::fromBase64() should return static(Laragear\WebAuthn\ByteBuffer) but
         returns Laragear\WebAuthn\ByteBuffer.
 ------ -------------------------------------------------------------------------------------------------------------

 ------ -------------------------------------------------------------------------
  Line   CborDecoder.php
 ------ -------------------------------------------------------------------------
  143    Dead catch - InvalidArgumentException is never thrown in the try block.
 ------ -------------------------------------------------------------------------

 ------ ------------------------------------------------------------------------------------------------------------
  Line   Exceptions/AssertionException.php
 ------ ------------------------------------------------------------------------------------------------------------
  16     PHPDoc tag @return with type Laragear\WebAuthn\Exceptions\AssertionException is not subtype of native type
         static(Laragear\WebAuthn\Exceptions\AssertionException).
 ------ ------------------------------------------------------------------------------------------------------------

 ------ --------------------------------------------------------------------------------------------------------------
  Line   Exceptions/AttestationException.php
 ------ --------------------------------------------------------------------------------------------------------------
  16     PHPDoc tag @return with type Laragear\WebAuthn\Exceptions\AttestationException is not subtype of native type
         static(Laragear\WebAuthn\Exceptions\AttestationException).
 ------ --------------------------------------------------------------------------------------------------------------

 ------ ----------------------------------------------------------------------------------------------------------
  Line   Http/Requests/AssertedRequest.php
 ------ ----------------------------------------------------------------------------------------------------------
  60     Method Laragear\WebAuthn\Http\Requests\AssertedRequest::login() should return
         (Illuminate\Contracts\Auth\Authenticatable&Laragear\WebAuthn\Contracts\WebAuthnAuthenticatable)|null but
         returns Illuminate\Foundation\Auth\User|null.
 ------ ----------------------------------------------------------------------------------------------------------

 ------ -------------------------------------------------------------------------------------
  Line   Http/Requests/AssertionRequest.php
 ------ -------------------------------------------------------------------------------------
  136    Call to an undefined method Illuminate\Contracts\Auth\StatefulGuard::getProvider().
  137    Call to an undefined method Illuminate\Contracts\Auth\StatefulGuard::getProvider().
 ------ -------------------------------------------------------------------------------------

 ------ -----------------------------------------------------------------------------------------------------------
  Line   Http/Requests/AttestationRequest.php
 ------ -----------------------------------------------------------------------------------------------------------
  55     Parameter #1 $user of class Laragear\WebAuthn\Attestation\Creator\AttestationCreation constructor expects
         Laragear\WebAuthn\Contracts\WebAuthnAuthenticatable, Illuminate\Foundation\Auth\User|null given.
 ------ -----------------------------------------------------------------------------------------------------------

 ------ ---------------------------------------------------------------------------------------------------------------
  Line   Http/Requests/AttestedRequest.php
 ------ ---------------------------------------------------------------------------------------------------------------
  67     Parameter #1 $user of class Laragear\WebAuthn\Attestation\Validator\AttestationValidation constructor expects
         Laragear\WebAuthn\Contracts\WebAuthnAuthenticatable, Illuminate\Foundation\Auth\User|null given.
  85     Parameter #1 $user of event class Laragear\WebAuthn\Events\CredentialCreated constructor expects
         Laragear\WebAuthn\Contracts\WebAuthnAuthenticatable in
         Laragear\WebAuthn\Events\CredentialCreated::dispatch(), Illuminate\Foundation\Auth\User|null given.
 ------ ---------------------------------------------------------------------------------------------------------------

 ------ --------------------------------------------------------------------------------------------------------------
  Line   Models/WebAuthnCredential.php
 ------ --------------------------------------------------------------------------------------------------------------
  103    Type Laragear\WebAuthn\Contracts\WebAuthnAuthenticatable in generic type
         Illuminate\Database\Eloquent\Relations\MorphTo<Laragear\WebAuthn\Contracts\WebAuthnAuthenticatable,
         Laragear\WebAuthn\Models\WebAuthnCredential> in PHPDoc tag @return is not subtype of template type
         TRelatedModel of Illuminate\Database\Eloquent\Model of class Illuminate\Database\Eloquent\Relations\MorphTo.
  105    Method Laragear\WebAuthn\Models\WebAuthnCredential::authenticatable() should return
         Illuminate\Database\Eloquent\Relations\MorphTo<Laragear\WebAuthn\Contracts\WebAuthnAuthenticatable,
         Laragear\WebAuthn\Models\WebAuthnCredential> but returns
         Illuminate\Database\Eloquent\Relations\MorphTo<Illuminate\Database\Eloquent\Model,
         Laragear\WebAuthn\Models\WebAuthnCredential>.
 ------ --------------------------------------------------------------------------------------------------------------
```

</details>

**Comment by DarkGhostHunter** [comment]
I know PHPStan. No problem, since it may help me to maintain this, but first test must pass.

**Comment by szepeviktor** [comment]
> test must pass

We could do that!

File: src/Models/WebAuthnCredential.php [hunk 1]
```diff
@@ -98,7 +98,7 @@ class WebAuthnCredential extends Model
     protected $visible = ['id', 'origin', 'alias', 'aaguid', 'attestation_format', 'disabled_at', 'is_enabled'];
 
     /**
-     * @return \Illuminate\Database\Eloquent\Relations\MorphTo&\Laragear\WebAuthn\Contracts\WebAuthnAuthenticatable
+     * @return \Illuminate\Database\Eloquent\Relations\MorphTo<\Laragear\WebAuthn\Contracts\WebAuthnAuthenticatable, \Laragear\WebAuthn\Models\WebAuthnCredential>
```
Comment (szepeviktor) [inline src/Models/WebAuthnCredential.php:101]: AFAIK `MorphTo` needs two Models.
https://github.com/nunomaduro/larastan/blob/f72a8b1dfaa2eeaf746a93b3257591090c668794/stubs/MorphTo.stub#L6-L8

But `WebAuthnAuthenticatable` is not a Model.
What do you think?

Comment (DarkGhostHunter) [reply src/Models/WebAuthnCredential.php:101]: `WebAuthnAuthenticatable|Model`

Comment (szepeviktor) [reply src/Models/WebAuthnCredential.php:101]: I seee!!
So it is a "future" model that implements `WebAuthnAuthenticatable`. We do not know its name :(

**Comment by DarkGhostHunter** [comment]
You can hit a PR again once the changes have been made successfully.

**Pull request closed without merging**